WORK IN PROGRESS - Solidity standalone, with soltest-via-IPC merged

bobsummerwill opened the pull request:
Replaces https://github.com/ethereum/solidity/pull/525.

bobsummerwill committed:
Drop full copies of cmake/ and libdevcore/ into solidity repo.
I'll get that working, so we have a standalone, repository, and will then trim it back from there.

bobsummerwill committed:
Add gitignore entry for CLion IDE temporary files (.idea)

bobsummerwill committed:
Ugly first pass, which builds.

bobsummerwill committed:
Deleted a bunch of files from our orphaned copy of libdevcore which we don't use.
These were the easy cuts.

bobsummerwill committed:
Minimize CommonIO.

bobsummerwill committed:
Stripped out all the excess cmake helper files.

bobsummerwill committed:
Chop out more stuff.   Looks like we're not doing any database stuff in solidity.

bobsummerwill committed:
Fixed relative path to self.

bobsummerwill committed:
More cleanup, of WindowsSDK and self-dependencies.

bobsummerwill committed:
Fixed whitespace in gitignore, which was missing line ending at end of file.

bobsummerwill committed:
Comment about the problem we have with extdep pattern when we have standalone repos.
Time for chocolatey right now?

bobsummerwill committed:
Re-added pthread dependency for Linux builds of solidity.
It was previously "not MSVC", but it looks like we only need it for Linux, not for OS X.
The same fix might even be possible upstream.   I should try that.

bobsummerwill committed:
Chopped out most of the code from RLP.

Our only remaining API is ...

 inline bytes rlpList() { return RLPStream(0).out(); }

And the rest of internal code, which can be much simplified, because it's just going to be the result of that function which will be a constant.

bobsummerwill committed:
Shove all the RLP code into the source file.

bobsummerwill committed:
Chopped out almost all of the logging implementation (but not the API).

bobsummerwill committed:
Chopped out unused base36 code.

bobsummerwill committed:
Delete CommonJS, which appears not to be used.

bobsummerwill committed:
Only a little bit of CommonIO has strippable.
Lots of inlines.

bobsummerwill committed:
Chopped out a lot of Common.cpp/Common.h.
Most of what is left is pretty bad, IMHO.
Having global typedefs is not a good pattern in the first place, and the tensity of the names just makes these a maintenance issue.
Would probably be better to push these types up into the code which uses them?

bobsummerwill committed:
Chop out logging.   OK as per Christian.

bobsummerwill committed:
Worked through CommonData, most of which appears to be used.

bobsummerwill committed:
First pass at replicating the TravisCI, CircleCI and Appveyor automation I did for cpp-ethereum for solidity.

bobsummerwill committed:
Removed stray trailing backslash.

bobsummerwill committed:
Added "extdep" to standalone solidity.
We cannot actually have these "fake Windows packages" in both places, but this can be a workaround for now.

bobsummerwill committed:
OK - let's start with the full setup.sh from cpp-ethereum, and then chop it down from there.

bobsummerwill committed:
Try to minimize solidity setup.sh dependencies.

bobsummerwill committed:
Forgot to remove libjson-rpc-cpp from the list of dependencies on OS X.

bobsummerwill committed:
Removed curl from Linux setup (only used for RPC).

bobsummerwill committed:
Hmm.  Maybe we don't need CryptoPP or GMP either?

bobsummerwill committed:
Refreshed the CircleCI, TravisCI and Appveyor yml configuration files for standalone Solidity.
Bringing them up-to-date with all the progress made in cpp-ethereum, including switching to VS2015.
Added "deps" sub-module.

bobsummerwill committed:
Flush relevant webthree-helpers changes across into the orphaned "cmake" folder inside solidity standalone.
This should fix the paths such that VS2015 starts working.   I hope!

bobsummerwill committed:
Copied the full setup.sh from cpp-ethereum.
Want to get CircleCI builds working here, and then I will strip it back down again.
Might just be a caching issue?

bobsummerwill committed:
Stripped back out some WindowsSDK and Qt bits in the cmake files which we definitely don't need here for Solidity.

bobsummerwill committed:
Touch Solidity, to trigger CircleCI build.
The setup.sh still needs stripping right back.

bobsummerwill committed:
Stripped setup.sh back to just what it should be.
It looks like the remaining CircleCI issues are just some nasty caching thing.
Which may or may not be fixable.

bobsummerwill committed:
Deleted CircleCI support.
It is proving too unreliable to be worthwhile to maintain.
Caching issues.   No "matrix" ability.

bobsummerwill committed:
Strip LLVM PPA, which we don't need for Solidity.
It has actually been taken down temporarily too, but that just highlighted that we shouldn't have it here anyway.

bobsummerwill committed:
Merge branch 'develop' into solidity_standalone

# Conflicts:
#	.travis.yml

bobsummerwill committed:
Delete extdep folder, which is redundant now.
It was only needed for VS2013.

bobsummerwill committed:
Merge branch 'develop' into standalone_prior_to_splat

# Conflicts:
#	.travis.yml

bobsummerwill committed:
Added gitignore entry for Visual Studio Code VC browse temp file.

bobsummerwill committed:
Started work on re-enabling the tests.

bobsummerwill committed:
Pulled libethcore files into Solidity, along with a handful of other files:

- libdevcrypto/Common.h
- libevmcore/EVMSchedule.h

Trimming to be done here later.

bobsummerwill committed:
Drag all the dependencies in again, via libethereum headers in SolidityExecutionFramework.h.

    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:990:38: error:
          implicit instantiation of undefined template
          'std::__1::hash<boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<256,

Which seems to be a known issue whether there isn't a default hash function for multi precision numbers.

This is somehow working in the normal setup, presumably because of some overloaded function somewhere which we are missing.

Got to chop this stuff off way earlier.    We must have still been getting those libethereum dependencies indirectly elsewhere in the CMake setup.

bobsummerwill committed:
Stripped down to just Debug Trusty and El Capitan.
Updated Travis steps to account for the fact that we're not using the umbrella anymore.

bobsummerwill committed:
Hurrah!   Managed to strip libethereum out entirely.
Re-added the cache OPTIONS support to the travis batch file.

bobsummerwill committed:
Copying my minimal setup.sh content on top of install_dependencies.sh, but with eth and phoenix_sphinx added.
This is likely minimal, unless soltest needs more than my earlier branch.

bobsummerwill committed:
Attempting to get the tests working on Mac too.
Having to re-add leveldb dependency due to incomplete dependency minimizations.

bobsummerwill committed:
Stripped accidental leveldb dependency back out again, plus some headers.
Fixed incorrect ZIP path for macOS eth download/unzip.

bobsummerwill committed:
Had my executable names the wrong way around for macOS versus Linux.

bobsummerwill committed:
OK - let's NOT try to run the soltest for macOS for now, so we can get green and work from there.

bobsummerwill committed:
Fixed syntax error.

bobsummerwill committed:
Attempt to re-add Solidity test runs for OS X.

bobsummerwill committed:
Explicitly set extract directory for macOS unzip to ~.
I think the ethbin/ files are currently going somewhere weird and Travis-y.

bobsummerwill committed:
Using an explicit cd.  Wanted to avoid this, but want it working more.

bobsummerwill committed:
Disabled Solidity test builds on Windows because Tests-over-IPC code is using UNIX sockets unconditionally at the time of writing.

bobsummerwill committed:
Merge branch 'develop' into standalone_prior_to_splat

bobsummerwill committed:
Renaming of scripts, to match cpp-ethereum ones.

bobsummerwill committed:
Refresh Solidity standalone TravisCI and Appveyor setups from cpp-ethereum.
Will edit and iterate these from here.

bobsummerwill committed:
Stripped down install_deps.sh to just the subset of packages needed for solidity.
Updated all the comments to solidity rather than cpp-ethereum.

bobsummerwill committed:
Fix typo.

bobsummerwill committed:
Have to disable Solidity tests for Appveyor build, because of the work TODO on getting IPC working for Windows.

bobsummerwill committed:
OK - Let's try generating a Windows ZIP for Solidity.

bobsummerwill committed:
Added some pretty ugly parameterization into the Travis matrix for Solidity, to avoid pathetic little shell script just to do platform conditionals.
This probably isn't what we want to do, but I just want to iterate within Travis for now.

bobsummerwill committed:
soltest.exe isn't build able yet, so we cannot ZIP that :-)

bobsummerwill committed:
Commented out the rest of the Appveyor tests for Solidity, which don't work yet.

bobsummerwill committed:
Path tweaks for TravisCI and Appveyor builds.

bobsummerwill committed:
Travis path mixups.

bobsummerwill committed:
Try to fix path issues for TravisCI and Homebrew hack for eth install.

bobsummerwill committed:
Try to fix broken TravisCI ZIP step.

bobsummerwill committed:
Hmm - no config folder on macOS and Linux, it seems.

chriseth commented on .travis.yml:
```diff
@@ -6,10 +22,56 @@ matrix:
           dist: trusty
           sudo: required
           env: TRAVIS_BUILD_TYPE=Debug
+          env: CMAKE_OPTIONS="-DCMAKE_C_COMPILER=/usr/lib/ccache/$CC -DCMAKE_CXX_COMPILER=/usr/lib/ccache/$CXX"
```
I wonder whether this is needed given cmake's support for ccache

bobsummerwill replied:
Yeah - I am not sure on that either.   I've removed it since.   We can play around with it more in the future.

chriseth commented on .travis.yml:
```diff
@@ -6,10 +22,56 @@ matrix:
+    - ./install_deps.sh
+before_script:
+    - cd ..
+    - git clone https://github.com/ethereum/tests.git
```
This should not be needed, those are only the consensus tests

bobsummerwill replied:
Right - thanks - removed since.

chriseth commented on .travis.yml:
```diff
@@ -6,10 +22,56 @@ matrix:
+before_install:
+    - ./install_deps.sh
+before_script:
+    - cd ..
```
AFAIK, if you use this dash-syntax, it will just "resume next" on a failed command which is not really what we want.

bobsummerwill replied:
I've since done various restructuring into better "phases".   My understand is that even where it continues with the following steps, the phases and build as a whole will still fail, so it's just doing some extra work.    Again, we can iterate more on this later.  It's maybe just a bit sub-optimal now, but there are much worse elements than these in the slow-downs.

chriseth commented on .travis.yml:
```diff
@@ -6,10 +22,56 @@ matrix:
+    # Run the tests
+    #- $ETH_PATH --test -d /tmp/test
+    #- while [ ! -S /tmp/test/geth.ipc ]; do sleep 2; done
+    #- ./test/soltest --ipc /tmp/test/geth.ipc
```
Those have to be re-enabled before the merge

bobsummerwill replied:
Yeah - working on that, and for OSX too.

bobsummerwill replied:
PS.   The Windows ones are completely missing still.

chriseth commented on cmake/EthBuildInfo.cmake:
```diff
@@ -0,0 +1,57 @@
+		-DETH_BUILD_TYPE="${_cmake_build_type}"
+		-DETH_BUILD_OS="${ETH_BUILD_OS}"
+		-DETH_BUILD_COMPILER="${ETH_BUILD_COMPILER}"
+		-DETH_BUILD_JIT_MODE="${ETH_BUILD_JIT_MODE}"
```
Any reference to JIT should be removed from this repo

bobsummerwill replied:
Yeah - I have a note to do more pruning.   There may or may not be time to do that before the initial release.

chriseth commented on cmake/EthBuildInfo.cmake:
```diff
@@ -0,0 +1,57 @@
+		-DETH_BUILD_NUMBER="${BUILD_NUMBER}"
+		-DETH_VERSION_SUFFIX="${VERSION_SUFFIX}"
+		-DPROJECT_VERSION="${PROJECT_VERSION}"
+		-DETH_FATDB="${FATDB10}"
```
Fatdb is also not relevant to solidity

bobsummerwill replied:
As above.

chriseth commented on cmake/EthCompilerSettings.cmake:
```diff
@@ -0,0 +1,217 @@
+
+	# Configuration-specific compiler settings.	
+	set(CMAKE_CXX_FLAGS_DEBUG          "-O0 -g -DETH_DEBUG")
+	set(CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG -DETH_RELEASE")
```
Is NDEBUG used?

bobsummerwill replied:
Not sure.   There are plenty of pre-processor symbols which we could and should prune out.    Just not right now :-)

chriseth commented on cmake/EthDependencies.cmake:
```diff
@@ -0,0 +1,136 @@
+#message(STATUS "ctest path: ${CTEST_COMMAND}")
+
+## use multithreaded boost libraries, with -mt suffix
+set(Boost_USE_MULTITHREADED ON)
```
Not sure what this results in but I think it is not a good idea to have it for solidity.

bobsummerwill replied:
Yep - more pruning TODOs.

chriseth commented on cmake/EthExecutableHelper.cmake:
```diff
@@ -0,0 +1,315 @@
+	set (multi_value_args)
+	cmake_parse_arguments (ETH_INSTALL_EXECUTABLE "${options}" "${one_value_args}" "${multi_value_args}" "${extra_macro_args}")
+
+	if (ETH_INSTALL_EXECUTABLE_QMLDIR)
```
Anything related to QML can be wiped.

bobsummerwill replied:
I did a big clear-out of all that stuff upstream in webthree-helpers.    Have pulled that into cpp-ethereum.    Just need to echo that here too.    Again, it's pruning of now unused code-paths, not anything which is broken.

chriseth commented on cmake/EthExecutableHelper.cmake:
```diff
@@ -0,0 +1,315 @@
+	elseif (DEFINED MSVC)
+
+		get_target_property(TARGET_LIBS ${EXECUTABLE} INTERFACE_LINK_LIBRARIES)
+		string(REGEX MATCH "Qt5::Core" HAVE_QT ${TARGET_LIBS})
```
Qt can all go

chriseth commented on cmake/EthExecutableHelper.cmake:
```diff
@@ -0,0 +1,315 @@
+		set(CPACK_NSIS_URL_INFO_ABOUT "https://ethereum.org")
+		set(CPACK_NSIS_CONTACT "ethereum.org")
+		set(CPACK_NSIS_MODIFY_PATH OFF)
+		set(CPACK_NSIS_MUI_ICON "${CMAKE_CURRENT_SOURCE_DIR}/res/win/alethzero.ico")
```
Alethzero? Do we actually want an installer at all?

chriseth commented on cmake/EthExecutableHelper.cmake:
```diff
@@ -0,0 +1,315 @@
+endmacro()
+
+macro(eth_appdmg)
+	if (GUI AND DEFINED APPLE)
```
There is no GUI

bobsummerwill replied:
Yep - all gone upstream.   Just needs refreshing.

chriseth commented on cmake/EthExecutableHelper.cmake:
```diff
@@ -0,0 +1,315 @@
+	endif()
+endmacro()
+
+macro(jsonrpcstub_client_create SPEC CLIENTNAME CLIENTDIR CLIENTFILENAME)
```
No use of jsonrpc (the library, not the concept) inside solidity

bobsummerwill replied:
Right - just jsoncpp.

chriseth commented on cmake/EthOptions.cmake:
```diff
@@ -0,0 +1,143 @@
+	set(NAME ${PROJECT_NAME})
+
+	# features
+	eth_default_option(VMTRACE OFF)
```
All these options can go.

bobsummerwill replied:
Yep

chriseth commented on cmake/EthOptions.cmake:
```diff
@@ -0,0 +1,143 @@
+	eth_default_option(MINIUPNPC ON)
+
+	# components
+	eth_default_option(GUI ON)
```
These don't make sense either.

bobsummerwill replied:
Yep

chriseth commented on cmake/EthUtils.cmake:
```diff
@@ -0,0 +1,114 @@
+endmacro()
+
+# Creates C resources file from files
+function(eth_add_resources RESOURCE_FILE OUT_FILE ETH_RES_DIR)
```
No resources.

bobsummerwill replied:
Yep

chriseth commented on cmake/scripts/buildinfo.cmake:
```diff
@@ -0,0 +1,49 @@
+) 
+
+if (NOT ETH_COMMIT_HASH)
+	set(ETH_COMMIT_HASH 0)
```
Note: We will soon require this to fail because we want that hash to be stamped into the bytecode. Note that there we need a workaround for places where we build from source but where we don't have a `.git` directory like in the ppa build process.

chriseth commented on libdevcore/Diff.h:
```diff
@@ -0,0 +1,52 @@
+/*
```
I think this file can go.

chriseth commented on libdevcore/Exceptions.h:
```diff
@@ -0,0 +1,87 @@
+
+/// Base class for all RLP exceptions.
+struct RLPException: virtual Exception { RLPException(std::string _message = std::string()): Exception(_message) {} };
+#define DEV_SIMPLE_EXCEPTION_RLP(X) struct X: virtual RLPException { const char* what() const noexcept override { return #X; } }
```
rlp exceptions can go

chriseth commented on libdevcore/Exceptions.h:
```diff
@@ -0,0 +1,87 @@
+DEV_SIMPLE_EXCEPTION_RLP(UndersizeRLP);
+
+DEV_SIMPLE_EXCEPTION(BadHexCharacter);
+DEV_SIMPLE_EXCEPTION(NoNetworking);
```
Most of these are not needed

chriseth commented on libdevcore/Exceptions.h:
```diff
@@ -0,0 +1,87 @@
+
+// error information to be added to exceptions
+using errinfo_invalidSymbol = boost::error_info<struct tag_invalidSymbol, char>;
+using errinfo_wrongAddress = boost::error_info<struct tag_address, std::string>;
```
Most of these should be unnecessary

chriseth commented on libdevcore/FileSystem.h:
```diff
@@ -0,0 +1,39 @@
+/*
```
File can be deleted

chriseth commented on libdevcore/Guards.h:
```diff
@@ -0,0 +1,159 @@
+/*
```
No multithreading in solidity

bobsummerwill replied:
Something depends on it, I think.

chriseth commented on libdevcore/Hash.cpp:
```diff
@@ -0,0 +1,440 @@
+	return ret;
+}
+
+namespace rmd160
```
Not needed

bobsummerwill replied:
It was.   Not sure why.

chriseth commented on libdevcore/RLP.cpp:
```diff
@@ -0,0 +1,378 @@
+/*
```
Not needed

chriseth commented on libdevcore/RLP.h:
```diff
@@ -0,0 +1,487 @@
+/*
```
Not needed

bobsummerwill replied:
Looks like the whole of RLP is dragged in purely for initializing EmptyTrie constant!

What constant should I replace this with?

h256 const EmptyTrie = sha3(rlp(""));

chriseth commented on libdevcore/SHA3.h:
```diff
@@ -0,0 +1,73 @@
+
+/// Calculate SHA3-256 hash of the given input, returning as a 256-bit hash.
+inline h256 sha3(bytesConstRef _input) { h256 ret; sha3(_input, ret.ref()); return ret; }
+inline SecureFixedHash<32> sha3Secure(bytesConstRef _input) { SecureFixedHash<32> ret; sha3(_input, ret.writable().ref()); return ret; }
```
I think we need none of the "secure" stuff.

chriseth commented on libdevcore/Terminal.h:
```diff
@@ -0,0 +1,148 @@
+#pragma once
```
Not needed

bobsummerwill replied:
There is a physical dependency right now, I believe.

chriseth commented on libdevcore/Worker.h:
```diff
@@ -0,0 +1,107 @@
+/*
```
No worker / multithreading in solidity

bobsummerwill replied:
Something needed it.

chriseth commented on libdevcrypto/Common.h:
```diff
@@ -0,0 +1,226 @@
+/// @NOTE This is not endian-specific; it's just a bunch of bytes.
+using Signature = h520;
+
+struct SignatureStruct
```
Anything related to signatures can go, I think.

bobsummerwill replied:
Again - something needed it.

chriseth commented on libdevcrypto/Common.h:
```diff
@@ -0,0 +1,226 @@
+extern Address ZeroAddress;
+
+/// A vector of Ethereum addresses.
+using Addresses = h160s;
```
We sill need addresses

bobsummerwill replied:
Yep

chriseth commented on libdevcrypto/Common.h:
```diff
@@ -0,0 +1,226 @@
+bool decryptSym(Secret const& _k, bytesConstRef _cipher, bytes& o_plaintext);
+
+/// Encrypt payload using ECIES standard with AES128-CTR.
+void encryptECIES(Public const& _k, bytesConstRef _plain, bytes& o_cipher);
```
Encryption / decryption stuff can go

bobsummerwill replied:
Something depended on this, I think.

chriseth commented on libdevcrypto/Common.h:
```diff
@@ -0,0 +1,226 @@
+ * Thus, Nonce's primary intended use at this time is for networking 
+ * where the key is also stored in plaintext.
+ */
+class Nonce
```
No nonce

chriseth commented on libdevcrypto/Common.h:
```diff
@@ -0,0 +1,226 @@
+DEV_SIMPLE_EXCEPTION(InvalidState);
+
+/// Key derivation
+h256 kdf(Secret const& _priv, h256 const& _hash);
```
Not needed

chriseth commented on libdevcrypto/Common.h:
```diff
@@ -0,0 +1,226 @@
+/// Simple class that represents a "key pair".
+/// All of the data of the class can be regenerated from the secret key (m_secret) alone.
+/// Actually stores a tuplet of secret, public and address (the right 160-bits of the public).
+class KeyPair
```
Not needed

chriseth commented on libethcore/ABI.h:
```diff
@@ -0,0 +1,100 @@
+/*
```
I think this file can go

chriseth commented on libethcore/BasicAuthority.h:
```diff
@@ -0,0 +1,60 @@
+/*
```
The whole ethcore directory should not be needed - at least I made a refactoring that removed the dependency

bobsummerwill replied:
It's all indirect dependencies, probably because of the still unresolved EVM bits.

chriseth commented on libevm/ExtVMFace.h:
```diff
@@ -0,0 +1,303 @@
+/*
```
This directory should not be needed either.

bobsummerwill replied:
Yeah - it is needed because of the circular dependencies.

As per https://github.com/ethereum/webthree-umbrella/issues/137

chriseth commented on libevmcore/EVMSchedule.h:
```diff
@@ -0,0 +1,73 @@
+namespace eth
+{
+
+struct EVMSchedule
```
This is the only thing we need from the whole ethcore / evm cruft. Please make sure to use the new version by greg that uses unsigned instead of u256. If that would be a single file in this directory, you can put it inside libevmasm.

bobsummerwill replied:
Right.

chriseth commented on libevmcore/Instruction.h:
```diff
@@ -0,0 +1,265 @@
+/*
```
We already have a copy of that file in libevmasm.

bobsummerwill replied:
OK.

chriseth commented on test/libsolidity/SolidityExecutionFramework.cpp:
```diff
@@ -56,8 +61,8 @@ ExecutionFramework::ExecutionFramework() :
 	m_rpc(RPCSession::instance(getIPCSocketPath())),
 	m_sender(m_rpc.account(0))
 {
-	if (g_logVerbosity != -1)
```
`g_logVerbosity` can go altogether - no logging in Solidity.

chriseth commented on test/libsolidity/SolidityExecutionFramework.h:
```diff
@@ -22,20 +22,18 @@
 #include <fstream>
 #include "../TestHelper.h"
 #include "../RPCSession.h"
 #include <libethcore/ABI.h>
```
We should not need any of those, please check what is needed from them.

bobsummerwill committed:
Complete revamp of Solidity standalone Travis file.
Adding skeleton for release flow.
Simplified all the matrix stuff, modelling after Pawel's evmjit pattern.
Stripped cache stuff out, and sudo.

bobsummerwill committed:
Ran travis setup releases on it, to get a specific token set up.
Also does funny reformatting.

bobsummerwill committed:
Re-tabbed it back out, and readded the matrix element.

bobsummerwill committed:
More matrix tweaks.

bobsummerwill committed:
Refine ZIP generation and remove some probably unnecessary current directory changes.

bobsummerwill committed:
Stripped one two many of the cds.

bobsummerwill committed:
Stripped ETHEREUM_TEST_PATH stuff, which is for the EVM.
Not needed for Solidity.

bobsummerwill committed:
Double expression expansion tweaks for TravisCI.

bobsummerwill committed:
Minor updates to Solidity TravisCI/Appveyor setup, copying patterns from cpp-ethereum.
Tests are still disabled/broken, as are docs.

bobsummerwill committed:
Re-added the Solidity unit-tests for macOS and Trusty, and the Sphinx documentation.
This is a cleaner approach than the matrix one which proceeded it, though this stuff might be better in bash scripts.
Let's get it working again in a potentially committable form first, though.

bobsummerwill committed:
Divided up the Solidity automation across the different sections of TravisCI and Appveyor again, to try to get them as idiomatic as possible.

bobsummerwill committed:
Say "development build", not "development release", because it isn't a real release.
Also, flipped them to "pre-release" in Appveyor.

bobsummerwill committed:
OK - seems like we do need that explicit submodule update.   Weird.

bobsummerwill committed:
Corrected ZIP suffixes, so we have a ZIP for Yosemite and one for El Capitan.
Previously it was just based on the OS, and these would have had clashing names.

bobsummerwill committed:
That was bad.   Let's go back to a -make j2.

bobsummerwill committed:
Bash horror for double-expansion of expressions.

bobsummerwill committed:
Added 'eth' installation on Ubuntu, for the unit-tests-over-IPC.

bobsummerwill committed:
Working directory fixes for eth on macOS for Solidity tests.

bobsummerwill committed:
See if I can fix the paths for macOS tests for Solidity.   Grr.

bobsummerwill committed:
Install "cpp-ethereum" to get "eth" for the Ubuntu Solidity tests.

bobsummerwill committed:
Another macOS path fix for eth location in Solidity tests.

bobsummerwill committed:
Yuck.   More Homebrew installs required for that eth to work on macOS for the Solidity tests.

bobsummerwill committed:
Output dir for ZIP to root.

bobsummerwill committed:
Travis gmp Homebrew workaround.

bobsummerwill committed:
Switched to Yosemite eth binary.

bobsummerwill committed:
Added some debug strings describing the hardware we're running on for macOS in Travis.

bobsummerwill committed:
eth is just in the ethereum-dev PPA?

bobsummerwill committed:
We need both PPAs.

bobsummerwill committed:
Hitting no networking issue

bobsummerwill committed:
Switched to a Mavericks cpp-ethereum ZIP from github, and added -L to follow redirects.

bobsummerwill committed:
OK - weird directory structure for the cpp-ethereum-develop-osx-mavericks ZIP.   Whatever.

bobsummerwill committed:
Typo in ZIP name.

bobsummerwill committed:
Changed comment-out code for g_logVerbosity into plain deletions.

bobsummerwill committed:
Debugging Mavericks ZIP

bobsummerwill committed:
Stripped out libevm files.

bobsummerwill committed:
Chopped out a bunch of header includes from SolidityExecutionFramework.

bobsummerwill committed:
Chopped out all of libevmcore except EVMSchedule.h.

bobsummerwill committed:
Stripped down EVMSchedule, flipped to unsigned, as per Greg's changes.
Moved it into evmcoreasm, per Christian's suggestion.

bobsummerwill committed:
Chopped out nearly all of libethcore.
The only bits left are some currency unit constants in Common.h and some ABI stuff used in SolidityExecutionFramework.

bobsummerwill committed:
Serious chunk of trimming.

bobsummerwill committed:
Deleted Diff.h, which is unused.

bobsummerwill committed:
Moved libdevcrypto and libethcore elements into libdevcore and/or inlining, so we're only added one new folder.

bobsummerwill committed:
Removed some debug text from the TravisCI file.

bobsummerwill committed:
Update Mavericks ZIP used for Solidity tests on Mac.
That one has got dylib paths fixed.

bobsummerwill committed:
Removed various hacks which we don't need anymore for macOS Solidity-test-over-IPC.
The Mavericks ZIP is very close to correct now.

bobsummerwill committed:
Using latest Mavericks ZIP, which should hopefully have all the correct dylibs in it now, with correct symbolic paths.

bobsummerwill committed:
Pick up latest ZIP, with lib snappy.dylib fix.

bobsummerwill committed:
Another ZIP update, to try to fix the leveldb -> snappy dependency to be self-contained.

bobsummerwill committed:
Get rid of bootstrap parameter and re-add trailing &

bobsummerwill committed:
Use environment variable to set IPC path for Solidity-tests-over-IPC.
There is something a bit wonky with the Boost tests configuration.
I had to do this on Appveyor too, in earlier attempts.

bobsummerwill committed:
Roll lines together.

bobsummerwill committed:
Comment out Solidity tests, so I can debug the ZIP generation.

bobsummerwill committed:
Need some commands, to avoid use of the default.

bobsummerwill committed:
ls for debugging.

bobsummerwill committed:
Re-added unit-test calls.
Absolute path for ZIP to deploy.

bobsummerwill committed:
Tweaks to RPCSession use of UNIX Sockets, to make them more portable.
This seems to get us working on macOS.

bobsummerwill committed:
Looks like sun_len member of sockaddr_un is platform-specific.
Needed on macOS.  Not even present on Ubuntu.

bobsummerwill committed:
RPC, and added a chunk of comment.

bobsummerwill committed:
Moved install_deps.sh and install_deps.bat into a scripts/ sub-folder, rather than polluting the root directory.

bobsummerwill committed:
Switched Solidity automation to build RelWithDebInfo, and also enabled Mavericks, because we can now, so why not!

bobsummerwill committed:
Updated install_deps.sh so it works for OS X Mavericks too.  Removed RLP dependencies, replacing EmptyTrie constant with a constant expression.

bobsummerwill committed:
Pull across ccache and OS X Mavericks support from cpp-ethereum.

bobsummerwill committed:
Full pass on cmake files, stripping out nearly all the options.
There is likely further fat to be trimmed here, but not today:
- CMakeParseArguments.cmake
- EthUtils.cmake
- FindPackageHandleStandardArgs.cmake
- FindPackageMessage.cmake
- FindSolidity.cmake
- UseDev.cmake
- UseSolidity.cmake

bobsummerwill committed:
Mirrored the release.bat / release.sh pattern from cpp-ethereum.
Adding dylib path fixup for macOS builds.

bobsummerwill committed:
Corrected relative path to TravisCI script for release.sh.

bobsummerwill committed:
Removed comment about boost.
We are statically linking against boost, so don't need to do anything with dylibs.

bobsummerwill committed:
Fixed batch typo.

bobsummerwill committed:
Try again to filter out the weird "untagged-XXXX" builds which keep popping up.

bobsummerwill committed:
Percentage sign for batch file parameters, not dollars (as is the case on UNIX).

bobsummerwill committed:
Enabled Appveyor and TravisCI caching.
Updated version to 0.3.6.

bobsummerwill committed:
Get cached version working.

bobsummerwill committed:
if not exist build mkdir build

chriseth commented:
There are two errors in the travis log which should mark this PR as failed but don't (only looking at ubuntu log):
https://travis-ci.org/ethereum/solidity/jobs/145735349#L1228
https://travis-ci.org/ethereum/solidity/jobs/145735349#L1329

chriseth commented on .travis.yml:
```diff
@@ -1,15 +1,77 @@
-cache: ccache
+branches:
+    except:
+        - /develop-v[0-9]/
```
Will this stay?

chriseth commented on .travis.yml:
```diff
@@ -1,15 +1,77 @@
+        - /develop-v[0-9]/
+        - /untagged-[0-9]/
 matrix:
     include:
```
You could add another matrix job only for the documentation - then it is only run once and in parallel.

chriseth commented on .travis.yml:
```diff
@@ -1,15 +1,77 @@
+install:
+    - ./scripts/install_deps.sh
+before_script:
+    - mkdir -p build
```
This is quite noisy in the logs. What about:

```
 - mkdir -p build && cd build && cmake .. -DCMAKE_BUILD_TYPE=$TRAVIS_BUILD_TYPE
 - make -j2
 - ../scripts/release.sh $ZIP_SUFFIX
```

chriseth commented on .travis.yml:
```diff
@@ -1,15 +1,77 @@
+    - make -j2
+    - ../scripts/release.sh $ZIP_SUFFIX
+script:
+    - ETH_PATH_FOR_linux="eth"
```
This should really be its own script.

chriseth commented on .travis.yml:
```diff
@@ -1,15 +1,77 @@
+    - export ETH_TEST_IPC=/tmp/test/geth.ipc && ./test/soltest
+    - pkill eth
+after_success:
+    - DOCS_COMMAND_FOR_linux="cd docs && sphinx-build -nW -b html -d _build/doctrees . _build/html && cd .."
```
Also: separate script

chriseth commented on CMakeLists.txt:
```diff
@@ -1,14 +1,14 @@
 
 # project name and version should be set after cmake_policy CMP0048
-set(PROJECT_VERSION "0.3.5")
+set(PROJECT_VERSION "0.3.6")
```
Let's make a new PR that only increments the version number.

chriseth commented on cmake/EthExecutableHelper.cmake:
```diff
@@ -0,0 +1,185 @@
+	endif()
+endmacro()
+
+macro(jsonrpcstub_client_create SPEC CLIENTNAME CLIENTDIR CLIENTFILENAME)
```
No jsonrpc

chriseth commented on cmake/EthOptions.cmake:
```diff
@@ -0,0 +1,40 @@
+
+	# components
+	eth_default_option(TESTS ON)
+	eth_default_option(TOOLS ON)
```
No tools

chriseth commented on cmake/EthOptions.cmake:
```diff
@@ -0,0 +1,40 @@
+if (SUPPORT_TESTS)
+	message("-- TESTS            Build tests                              ${TESTS}")
+endif()
+if (SUPPORT_TOOLS)
```
No tools

chriseth commented on libdevcore/ABI.h:
```diff
@@ -0,0 +1,100 @@
+template <> struct ABISerialiser<u256> { static bytes serialise(u256 const& _t) { return h256(_t).asBytes(); } };
+template <> struct ABISerialiser<u160> { static bytes serialise(u160 const& _t) { return bytes(12, 0) + h160(_t).asBytes(); } };
+template <> struct ABISerialiser<string32> { static bytes serialise(string32 const& _t) { bytes ret; bytesConstRef((byte const*)_t.data(), 32).populate(bytesRef(&ret)); return ret; } };
+template <> struct ABISerialiser<std::string>
```
This whole file should go.

chriseth commented on libdevcore/Base64.h:
```diff
@@ -0,0 +1,42 @@
+namespace dev
+{
+
+std::string toBase64(bytesConstRef _in);
```
Don't see any usage of this.

chriseth commented on libdevcore/Common.h:
```diff
@@ -0,0 +1,210 @@
+using bytesConstRef = vector_ref<byte const>;
+
+template <class T>
+class secure_vector
```
All the `secure_` stuff can probably go.

chriseth commented on libdevcore/Common.h:
```diff
@@ -0,0 +1,210 @@
+	std::function<void(void)> m_f;
+};
+
+enum class WithExisting: int
```
This can go.

chriseth commented on libdevcore/FixedHash.h:
```diff
@@ -0,0 +1,306 @@
+
+	/// Populate with random data.
+	template <class Engine>
+	void randomize(Engine& _eng)
```
I think we should not use any randomness inside solidity

chriseth commented on libdevcore/FixedHash.h:
```diff
@@ -0,0 +1,306 @@
+		return contains(_h.template bloomPart<P, N>());
+	}
+
+	template <unsigned P, unsigned M> inline FixedHash<M> bloomPart() const
```
Blooms can all go.

chriseth commented on libdevcore/vector_ref.h:
```diff
@@ -0,0 +1,120 @@
+	bool operator!=(vector_ref<_T> const& _cmp) const { return !operator==(_cmp); }
+
+#if DEV_LDB
+	operator ldb::Slice() const { return ldb::Slice((char const*)m_data, m_count * sizeof(_T)); }
```
Can be removed.

chriseth commented on test/libsolidity/SolidityEndToEndTest.cpp:
```diff
@@ -21,9 +21,25 @@
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#endif // defined(__GNUC__)
+
+#include <boost/thread.hpp>
```
Where is this needed? There is a sleep somewhere, but that should be taken from stdc++11, not from boost.

chriseth commented:
Almost there! :-)

bobsummerwill committed:
Switch to the released cpp-ethereum-v1.3.0 binary here for 'soltest'.
We can switch `brew install` when that is fixed up.

bobsummerwill committed:
Merge branch 'develop' into standalone_prior_to_splat
